test/ex_doc/formatter/html/erlang_test.exs

```diff
@@ -29,15 +29,33 @@ defmodule ExDoc.Formatter.HTML.ErlangTest do
 
     -type t() :: atom().
     %% t/0 type.
+
+    -record(rec, {k :: undefined}).
+
+    -type my_tea() :: #rec{k :: uri_string:uri_string() | undefined}.
+
+    -spec baz() -> my_tea().
+      baz() ->
+        Eh = <<"eh?">>,
+        #rec{k=Eh}.
     """)
 
-    doc = generate_docs(c)
+    refute ExUnit.CaptureIO.capture_io(:stderr, fn ->
+             send(self(), {:doc_path, generate_docs(c)})
+           end) =~ "inconsistency, please submit bug"
+
+    assert_received {:doc_path, doc}
```

Comment:
there's no need for this, if the warning is printed we'll (eventually) see it when running tests locally

```suggestion
    doc = generate_docs(c)
```

Reply:
I tried cherry-picking just this test on main and it works. I'd like to see it fail before committing the patch. Could you double-check? Perhaps Elixir or OTP version matters? What's yours?